config/config.go

```diff
@@ -417,6 +417,7 @@ type Config struct {
 	HTTPPorts       ListenConfig              `yaml:"httpPort"`
 	HTTPSPorts      ListenConfig              `yaml:"httpsPort"`
 	TLSPorts        ListenConfig              `yaml:"tlsPort"`
+	DoHUserAgent    string                    `yaml:"dohUserAgent" default:"blocky"`
```

Comment:
I thought it would be a free advertisement for the project (I guess), while still giving user the ability to override it.

@0xERR0R any thoughts on this?

Reply:
Just FYI, [RFC 8484](https://datatracker.ietf.org/doc/html/rfc8484#section-8.2) mentions that the HTTP's feature set like  User-Agent header can be used for identification and tracking, that's the original reason that I opened #446 and would like to do the same as iOS and Firefox ;)

>    HTTP's feature set can also be used for identification and tracking
>    in a number of different ways.  For example, Authentication request
>    header fields explicitly identify profiles in use, and HTTP cookies
>    are designed as an explicit state-tracking mechanism between the
>    client and serving site and often are used as an authentication
>    mechanism.
> 
>    Additionally, the User-Agent and Accept-Language request header
>    fields often convey specific information about the client version or
>    locale.  This facilitates content negotiation and operational work-
>    arounds for implementation bugs.  Request header fields that control
>    caching can expose state information about a subset of the client's
>    history.  Mixing DoH requests with other HTTP requests on the same
>    connection also provides an opportunity for richer data correlation.

Reply:
> I thought it would be a free advertisement for the project (I guess), while still giving user the ability to override it.
> 
> @0xERR0R any thoughts on this?

I would leave the default empty, we transmit only necessary information.

Reply:
Even if the advertisement thing should be done, it should be done in the server side, like using the "x-powered-by" header, otherwise I don't think advertise to those DoH service provider would good any good.